### PR TITLE
test(create): Add test to check deprecated `apiVersion`s in resource templates created by `helm create`

### LINKED ISSUE #11495: Helm create uses old HPA Version
[Jump to issue](#issue)

Output of `helm version`: 

```shell
➜ helm version    
version.BuildInfo{Version:"v3.9.4", GitCommit:"dbc6d8e20fe1d58d50e6ed30f09a04a77e4c68db", GitTreeState:"clean", GoVersion:"go1.17.13"}
```

Output of `kubectl version`:

```shell
➜ kubectl version                                    
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-12T10:57:26Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.3-gke.2100", GitCommit:"25d7334511e90d0b636707059c955baebce769cd", GitTreeState:"clean", BuildDate:"2022-08-16T09:24:54Z", GoVersion:"go1.18.3b7", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  GKE, v1.24.3-gke.2100

# Issue

When creating a new helm chart using `helm create <chart>` the HPA is using an old outdated and soon to be deprecated version

<details>
<summary>Warning</summary>

_note: Task is just a better version of makefile, the command it runs is `helm upgrade <reelase> <chart>

```shell
➜ task upgrade
task: [upgrade] helm upgrade $releaseName $chart
W1102 13:34:17.245653   42748 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2 HorizontalPodAutoscaler
W1102 13:34:17.269708   42748 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2 HorizontalPodAutoscaler
W1102 13:34:20.310690   42748 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2 HorizontalPodAutoscaler
W1102 13:34:20.438095   42748 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2 HorizontalPodAutoscaler
W1102 13:34:20.463040   42748 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2 HorizontalPodAutoscaler
W1102 13:34:20.488023   42748 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2 HorizontalPodAutoscaler

```
</details>

The Deprecation notice is [Here](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#horizontalpodautoscaler-v126)

What would be nice to see is the New version:

## Issues related

https://github.com/helm/helm/issues/10710

### PR DESCRIPTION
**Pull request by Bhargav-InfraCloud**
**What this PR does / why we need it**:
1. Change `HorizontalPodAutoscaler`'s version from `autoscaling/v2beta1` -> `autoscaling/v2` in `helm create`.
2. Add a test `TestHelmCreateChart_CheckDeprecatedWarnings` to check the deprecated versions from being created by `helm create`.
3. A comment is added in `pkg/chartutil/create.go` for future manifest files added to `helm create` to follow the same pattern and update the test.

About `TestHelmCreateChart_CheckDeprecatedWarnings`:
- Checks the manifests created by `helm create` to not have deprecated `apiVersion`s.
- This requires the following `ldflags` to be set per the current Kubernetes version for avoiding false positive results.
    1. `-X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMajor=`\<k8s-major-version>
    2. `-X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMinor=`\<k8s-minor-version>
- In order to avoid the `ldflags` passed for this test from conflicting with other tests, a seperate `go test` is added in `make test-unit`.
- Without the `ldflags`, this test passes with a false positive result. This behavior will help in the former run of `go test` and in `make test-coverage`. Note: This test doesn't contribute to coverage anyway.

Closes #11495 

**Special notes for your reviewer**:
Not sure how backward compatibility is checked when the changes are related to the `apiVersion` of the object. But else, I think this change qualifies as backward compatible. Please check.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

@userbradley @joejulian @hzliangbin @wujunwei Please validate whether these changes achieve the expected result. Thanks!

### CODE REVIEW

**Comment by userbradley** [comment]
ACK -  Will try tomorrow

**Review by joejulian — APPROVED** [review]
Looks good to me. :+1:

**Comment by vsroy** [comment]
@joejulian @Bhargav-InfraCloud 

Good Day.
Looks like Kubernetes v1.24 has `End of Life:2023-07-28`
https://kubernetes.io/releases/#release-v1-24

The `autoscaling/v2beta1` API version of HorizontalPodAutoscaler is no longer served as of  Kubernetes v1.25.
https://kubernetes.io/docs/reference/using-api/deprecation-guide/#horizontalpodautoscaler-v125

Can we merge this PR if it looks good ?

**Comment by joejulian** [comment]
@vsroy PRs larger than a small require two approvals.

**Comment by vsroy** [comment]
@hickeyma @jdolitsky @marckhouzam
Can we have a review on this please.
Based on this -> https://github.com/helm/helm/pull/11923#issuecomment-1566579532
cc @Bhargav-InfraCloud

**Comment by Vedrillan** [comment]
The API has changed between v2beta1 and v2. The  `specs.metrics[].Resource` needs to be updated as well if the manifest is upgraded to v2. In particular `targetAverageUtilization` no longer exists.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#resourcemetricsource-v2beta1-autoscaling
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#resourcemetricsource-v2-autoscaling

**Comment by Bhargav-InfraCloud** [comment]
Thanks! @Vedrillan @vsroy I've updated the PR. Please review.

**Review by mattfarina — APPROVED** [review]

**Pull request merged**
